### PR TITLE
feat: update CloudFormation spec to 10.5.0

### PR DESCRIPTION
**Pull request by rix0rrr**
----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

File: packages/@aws-cdk/aws-appconfig/package.json [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+  "jest": {
+    "moduleFileExtensions": [
+      "js"
+    ],
+    "coverageThreshold": {
+      "global": {
+        "branches": 60,
+        "statements": 80
+      }
+    }
```
Comment (nija-at) [inline packages/@aws-cdk/aws-appconfig/package.json]: It seems I missed to fix up the template that generates this. Can you point me at it?

Jest config [is now centralized](https://github.com/aws/aws-cdk/commit/2c5c88d33d51746eb35fe46068b0c5fd509d8d7a), so all is needed here is `"jest": {}` unless you want to override.

Comment (RomainMuller) [reply packages/@aws-cdk/aws-appconfig/package.json]: It's there:

https://github.com/aws/aws-cdk/blob/491e2d970c849cbc00e7cb3def927b12054d572f/packages/%40aws-cdk/cfnspec/build-tools/create-missing-libraries.ts#L170-L180

File: packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts [hunk 2]
```diff
@@ -89,7 +89,10 @@ async function main() {
       return;
     }
 
-    pushDownFirstAdditions(update);
+    pushDownCompleteChanges(update);
+    // tslint:disable-next-line:no-console
+    console.error(JSON.stringify(update, undefined, 2));
```
Comment (nija-at) [inline packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts]: Left behind from your local debugging?

File: packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts [hunk 3]
```diff
@@ -146,15 +149,38 @@ async function main() {
...
-  function pushDownFirstAdditions(update: any) {
-    for (const key of Object.keys(update)) {
-      const added = isAdded(key);
-      if (!added) { continue; }
-      const data = update[key];
-      delete update[key];
-      update[added] = {};
-      for (const subKey of Object.keys(data)) {
-        update[added][`${subKey}__added`] = data[subKey];
+  /**
```
Comment (nija-at) [inline packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts]: Over time, we've built up a lot of logic here in spec-diff.ts but it lacks unit tests. We should consider an item in the future to refactor and tests to it.

Not needed as part of this change.

File: packages/@aws-cdk/aws-appconfig/package.json (hunk 1 above)
Comment (nija-at) [reply packages/@aws-cdk/aws-appconfig/package.json]: Thanks!

File: packages/decdk/package.json [hunk 4]
```diff
@@ -165,7 +165,8 @@
     "jsii-reflect": "^0.22.0",
     "jsonschema": "^1.2.5",
     "yaml": "1.7.2",
-    "yargs": "^15.1.0"
+    "yargs": "^15.1.0",
+    "@aws-cdk/aws-appconfig": "1.23.0"
```
[the remark this answers is not in the document]
Comment (jogold) [reply packages/decdk/package.json]: I wonder why it's not sorted, I added this in #5329:
https://github.com/aws/aws-cdk/blob/491e2d970c849cbc00e7cb3def927b12054d572f/packages/%40aws-cdk/cfnspec/build-tools/create-missing-libraries.ts#L290-L301

File: packages/@aws-cdk/aws-appconfig/package.json (hunk 1 above)
Comment (rix0rrr) [reply packages/@aws-cdk/aws-appconfig/package.json]: Did it here.

File: packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts (hunk 3 above)
Comment (nija-at) [reply packages/@aws-cdk/cfnspec/build-tools/spec-diff.ts]: Opened an issue - https://github.com/aws/aws-cdk/issues/6198

**Pull request merged**